Add `Copy address` Peers Tab Context Menu Action

jarolrod opened the pull request:
Picking up #264 

This adds a `Copy Address` context menu action to the `Peers Tab`.

Based on the first commit of PR #317 so that we can use `Qt::DisplayRole` in the `copyEntryData` function.

| Master        | PR               |
| ----------- | ----------- |
|  ![Screen Shot 2021-05-05 at 4 51 11 AM](https://user-images.githubusercontent.com/23396902/117117822-fb067400-ad5d-11eb-9466-228456108e52.png) | ![Screen Shot 2021-05-05 at 4 49 15 AM](https://user-images.githubusercontent.com/23396902/117117835-fe99fb00-ad5d-11eb-8de0-f6a9acdbf40e.png) |

jarolrod commented:
updated from 6ee6d2f -> d6099b7 ([pr318.01](https://github.com/jarolrod/gui/tree/pr318.01), [pr318.02](https://github.com/jarolrod/gui/tree/pr318.02), [diff](https://github.com/jarolrod/gui/compare/pr318.01...jarolrod:pr318.02))

- rebase on top of updated last commit of #317 
- add translator comment for `Copy address` action

jarolrod commented:
updated from d6099b7 -> 85e0f95 ([pr318.02](https://github.com/jarolrod/gui/tree/pr318.02) -> [pr318.03](https://github.com/jarolrod/gui/tree/pr318.03))

**Changes:**
- Rebased over master because of conflict with #313 (because this is based on #317)

jarolrod commented:
updated from 85e0f95 -> 65d1d35 ([pr318.03](https://github.com/jarolrod/gui/tree/pr318.03) -> [pr318.04](https://github.com/jarolrod/gui/tree/pr318.04), [diff](https://github.com/jarolrod/gui/compare/pr318.03...jarolrod:pr318.04))

- rebased to solve conflict with #311 (because this is based on #317)

luke-jr commented on src/qt/rpcconsole.cpp:
```diff
@@ -647,6 +647,11 @@ void RPCConsole::setClientModel(ClientModel *model, int bestblock_height, int64_
         // create peer table context menu
         peersTableContextMenu = new QMenu(this);
+        //: Context menu action to copy the address of a peer
+        peersTableContextMenu->addAction(tr("Copy address"), [this] {
```
Please add a keyboard shortcut (&C used elsewhere)

jarolrod replied:
Thanks, updated

saadbitcoin commented:
Thank you

On Sun, 13 Jun 2021 at 00:01, Luke Dashjr ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/qt/rpcconsole.cpp
> <https://github.com/bitcoin-core/gui/pull/318#discussion_r650436039>:
>
> > @@ -647,6 +647,11 @@ void RPCConsole::setClientModel(ClientModel *model, int bestblock_height, int64_
>
>          // create peer table context menu
>          peersTableContextMenu = new QMenu(this);
> +        //: Context menu action to copy the address of a peer
> +        peersTableContextMenu->addAction(tr("Copy address"), [this] {
>
> Please add a keyboard shortcut (&C used elsewhere)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bitcoin-core/gui/pull/318#pullrequestreview-682354013>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHV67JUETFTOPZZYZLR4EWLTSPDKXANCNFSM44EISKIA>
> .
>

saadbitcoin commented:
OKY


On Mon, 14 Jun 2021 at 06:08, saad ***@***.***> wrote:

> Thank you
>
> On Sun, 13 Jun 2021 at 00:01, Luke Dashjr ***@***.***> wrote:
>
> > ***@***.**** commented on this pull request.
> > ------------------------------
> >
> > In src/qt/rpcconsole.cpp
> > <https://github.com/bitcoin-core/gui/pull/318#discussion_r650436039>:
> >
> > > @@ -647,6 +647,11 @@ void RPCConsole::setClientModel(ClientModel
> *model, int bestblock_height, int64_
> >
> > // create peer table context menu
> > peersTableContextMenu = new QMenu(this);
> > + //: Context menu action to copy the address of a peer
> > + peersTableContextMenu->addAction(tr("Copy address"), [this] {
> >
> > Please add a keyboard shortcut (&C used elsewhere)
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/bitcoin-core/gui/pull/318#pullrequestreview-682354013>,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AHV67JUETFTOPZZYZLR4EWLTSPDKXANCNFSM44EISKIA
> >
> > .
> >
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bitcoin-core/gui/pull/318#issuecomment-860334374>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHV67JVE3BILIESD2XR2ESDTSVXDRANCNFSM44EISKIA>
> .
>

promag commented:
@jarolrod needs rebase, this is a cool change.

jonatack commented:
Concept ACK

jarolrod commented:
updated from 65d1d35 -> 3ec061d ([pr318.04](https://github.com/jarolrod/gui/tree/pr318.04) -> [pr318.05](https://github.com/jarolrod/gui/tree/pr318.05))

Changes:
- rebased over recent changes
- Dropped commit from #317 as it was merged
- addressed @luke-jr comments and added a mnemonic shortcut

luke-jr reviewed (approved):
utACK 3ec061d9da0c8742bd9dec94ffeb82a11d000aba

hebasto commented on src/qt/rpcconsole.cpp:
```diff
@@ -680,6 +680,11 @@ void RPCConsole::setClientModel(ClientModel *model, int bestblock_height, int64_
 
         // create peer table context menu
         peersTableContextMenu = new QMenu(this);
+        //: Context menu action to copy the address of a peer
```
nit: 
```suggestion
        //: Context menu action to copy the address of a peer.
```

hebasto reviewed (approved):
ACK 3ec061d9da0c8742bd9dec94ffeb82a11d000aba, tested on Linux Mint 20.2 (Qt 5.12.8):

![Screenshot from 2021-09-12 18-49-30](https://user-images.githubusercontent.com/32963518/132994212-e0450618-3e23-4a60-9b45-9b506bcdb588.png)

jonatack commented:
Nice and useful change!
